### ISSUE TITLE
jest/no-if should not complain when not directly nested in `it`

### ISSUE THREAD

**Issue by lemonmade** [Shopify/eslint-plugin-shopify#263]
The following code currently warns, but shouldn't:

```ts
it('works', () => {
  const values = something.map((thing) => {
    if (thing.isFoo) {
      return thing.foo
    } else {
      return thing.bar;
    }
  });

  expect(values).toStrictEqual(['foo']);
});
```
